orangecontrib/prototypes/widgets/owimagegrid.py

```diff
@@ -0,0 +1,1141 @@
+import enum
+import itertools
+import logging
+import os
+from collections import namedtuple
+from concurrent.futures import Future
+from itertools import zip_longest
+
+import Orange.data
+import numpy as np
+from AnyQt.QtCore import (
+    Qt, QEvent, QSize, QSizeF, QRectF, QPointF, QUrl, QDir
+)
+from AnyQt.QtCore import pyqtSignal as Signal, pyqtSlot as Slot
+from AnyQt.QtGui import (
+    QPixmap, QPen, QBrush, QColor, QPainter, QImageReader
+)
+from AnyQt.QtWidgets import (
+    QGraphicsView, QGraphicsWidget, QGraphicsItem,
+    QGraphicsLinearLayout,
+    QGraphicsGridLayout, QSizePolicy, QApplication, QStyle, QShortcut,
+    QFormLayout)
+from Orange.widgets import widget, gui, settings
+from Orange.widgets.data.owimageviewer import ImageLoader, Preview, GraphicsScene
+from Orange.widgets.utils.itemmodels import VariableListModel
+from Orange.widgets.utils.overlay import proxydoc
+from Orange.widgets.widget import Input, Output
+
+from orangecontrib.prototypes.image_grid import ImageGrid
+
+_log = logging.getLogger(__name__)
+
+_ImageItem = namedtuple(
+    "_ImageItem",
+    ["index",  # Index in the input data table
+     "widget",  # GraphicsThumbnailWidget displaying the image.
+     "url",  # Composed final image url.
+     "future"]  # Future instance yielding an QImage
+)
+
+
+class OWImageGrid(widget.OWWidget):
+    name = "Image Grid"
+    description = "Visualize images in a similarity grid"
+    icon = "icons/Category-Prototypes.svg"
+    priority = 160
+    category = ""
+    keywords = ["image", "grid", "similarity"]
+    graph_name = "scene"
+
+    class Inputs:
+        data = Input("Embeddings", Orange.data.Table)
+        # data_subset = Input("Embeddings Subset", Orange.data.Table)
+
+    class Outputs:
+        data = Output("Images", Orange.data.Table)
+
+    _NO_DATA_INFO_TEXT = "No data on input."
+
+    settings_version = 2
+    settingsHandler = settings.DomainContextHandler()
+
+    cell_fit = settings.Setting("Resize")
+    columns = settings.Setting(10)
+    rows = settings.Setting(10)
+
+    imageAttr = settings.ContextSetting(0)
+    imageSize = settings.Setting(100)
+
+    auto_update = settings.Setting(True)
+    auto_commit = settings.Setting(True)
+
+    def __init__(self):
+        super().__init__()
+
+        self.grid = None
+
+        self.data = None
+        self.allAttrs = []
+        self.stringAttrs = []
+        self.domainAttrs = []
+
+        self.selectedIndices = []
+
+        #: List of _ImageItems
+        self.items = []
+
+        self._errcount = 0
+        self._successcount = 0
+
+        self.imageAttrCB = gui.comboBox(
+            self.controlArea, self, "imageAttr",
+            box="Image Filename Attribute",
+            tooltip="Attribute with image filenames",
+            callback=[self.clearScene, self.setupScene],
```

Comment:
This actually works? Who would have thought.

Reply:
Surprisingly, it does, this was already in Image Viewer.